backend/api/src/algolia/migration.rs

```diff
@@ -2,231 +2,135 @@ use algolia::{
     model::attribute::{Attribute, FacetAttribute, SearchableAttributes},
     request::SetSettings,
 };
-use futures::future::BoxFuture;
 
-#[derive(Copy, Clone, Debug)]
-pub enum ResyncKind {
-    None,
-    Complete,
-}
-
-fn bad_batch_object<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    _jig_index: &'a str,
-    _course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    Box::pin(async move {
-        client.delete_object(media_index, "batch").await?;
-        Ok(())
-    })
-}
-
-fn set_searchable_fields_v1<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_searchable_fields_v2`
-    empty(client, media_index, jig_index, course_index)
-}
+use sqlx::PgConnection;
 
-fn set_attributes_for_faceting_v1<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_attributes_for_faceting_v3`
-    empty(client, media_index, jig_index, course_index)
-}
+use hashfn::hashfn;
 
-fn set_searchable_fields_v2<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_searchable_fields_v3`
-    empty(client, media_index, jig_index, course_index)
-}
-fn set_attributes_for_faceting_v2<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_attributes_for_faceting_v3`
-    empty(client, media_index, jig_index, course_index)
-}
-
-fn set_attributes_for_faceting_v3<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_attributes_for_faceting_v4`
-    empty(client, media_index, jig_index, course_index)
-}
+const JIG_INDEX: &str = "jig_index";
+const MEDIA_INDEX: &str = "media_index";
+const COURSE_INDEX: &str = "course_index";
 
-fn set_attributes_for_faceting_v4<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_attributes_for_faceting_v5`
-    empty(client, media_index, jig_index, course_index)
-}
+pub(crate) const JIG_HASH: &'static str = JIG_INDEX_HASH;
+pub(crate) const MEDIA_HASH: &'static str = MEDIA_INDEX_HASH;
+pub(crate) const COURSE_HASH: &'static str = COURSE_INDEX_HASH;
```

Comment:
No need for these lines since you're using the generated name https://github.com/ji-devs/ji-cloud/pull/2739/files#diff-a730c27e0999c83f61e3f67674ead1ffc1c16c829ab6a30b751c2752222a435fR46. 

If you're setting them so it's clearer that these exist, I added a [change](https://github.com/johnnynotsolucky/hashfn/commit/061a68bbc9631ef10b4d3aee7e233b592873dfc2) which lets you optionally set the name of the const using the macro:

```rust
#[hashfn(MEDIA_HASH)]
pub(crate) async fn media_index(...) {
  // ...
}
```

and then using it later on

```rust
sqlx::query!(
  r#"update algolia_index_settings set updated_at = now(), index_hash = $1 where index_name = $2"#, 
  MEDIA_HASH, 
  MEDIA_INDEX
)
  .execute(txn)
  .await?;
```